File: Updater/converters/DerivedJoiningType.rb
=begin
 
 DerivedJoiningType.rb
   ©︎ 2018 YOCKOW.
     Licensed under MIT License.
     See "LICENSE.txt" for more information.
 
=end


module DerivedJoiningType
  PATH = 'Sources/UnicodeSupplement/__data__/UCD/DerivedJoiningType.swift'
  URLs = [
    URI.parse('https://www.unicode.org/Public/UCD/latest/ucd/extracted/DerivedJoiningType.txt')
  ]
end

module DerivedJoiningType; class << self
  def write(remote_io, file)
    table = Table.new(remote_io)
    failed("No data about DerivedJoiningType.") if table.count < 1
    
    jt_table = UnicodeTable.new(table, 1)
    file.write(jt_table.to_swift_assoc_array('jt', 'Character'))
  end
end; end

File: utils/SwiftUnicodeSupplementUpdater/Sources/UnicodeSupplementUpdater/Delegates/DerivedJoiningType.swift
/* *************************************************************************************************
 DerivedJoiningType.swift
   © 2020,2026 YOCKOW.
     Licensed under MIT License.
     See "LICENSE.txt" for more information.
 ************************************************************************************************ */
 
import Foundation
import UnicodeSupplement

public struct DerivedJoiningType: UCDDefaultablePropertiesCodeUpdaterDelegate {
  public typealias Property = Unicode.JoiningType

  public let dependencies: CodeDependencies = .init()

  public let setConversionCounter: ConversionCounter<String> = .init()

  public let dictionaryConversionCounter: ConversionCounter<String?> = .init()

  public init() {}

  public var prefix: String { return "jt" }

  public var sourceURLs: Array<URL> {
    return [
      URL(string: "https://www.unicode.org/Public/UCD/latest/ucd/extracted/DerivedJoiningType.txt")!
    ]
  }

  public func reduce(columns: [String]) throws -> Unicode.JoiningType {
    return Unicode.JoiningType(abbreviated: columns.first!)!
  }

  public func describe(value: Unicode.JoiningType) -> String {
    return ".\(String(describing: value))"
  }
  
  public var defaultValue: Unicode.JoiningType {
    return .nonJoining
  }
}


